engine/src/test/java/org/hibernate/search/testsupport/TestConstants.java

```diff
@@ -89,4 +85,11 @@ public static String getIndexDirectory(Class<?> testClass) {
 	public static boolean arePerformanceTestsEnabled() {
 		return Boolean.getBoolean( "org.hibernate.search.enable_performance_tests" );
 	}
+
+	private static class CallerProvider extends SecurityManager {
+
+		public Class<?> getCallerClass() {
+			return getClassContext()[2];
```

Comment:
How does the fork mode relate to this? This is just to obtain the calling class of the `getIndexDirectory()` method. How the class is loaded should not make a difference.